test/integration/Cloning.test.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  KitchenSink,
+  PrimitiveUrl,
+  CodeableConcept,
+  PrimitiveString,
+  Coding,
+  PrimitiveCode,
+  PrimitiveTime,
+  PrimitiveBoolean,
+  IElement,
+} from "../fixture/generatedTypeScript/fhir";
+
+describe("Test cloning of the generated classes", () => {
+  let sink: KitchenSink;
+
+  beforeEach(() => {
+    sink = new KitchenSink();
+    sink.system = "theSystem";
+    sink.url = PrimitiveUrl.parsePrimitive("theUrl");
+    sink.singleCode = new CodeableConcept();
+    sink.singleCode.text = PrimitiveString.parsePrimitive("theCode");
+    const coding1 = new Coding();
+    coding1.code = PrimitiveCode.parsePrimitive("code1");
+    const coding2 = new Coding();
+    coding2.code = PrimitiveCode.parsePrimitive("code2");
+    sink.coding = [coding1, coding2];
+    sink.times = [
+      PrimitiveTime.parsePrimitive("time1"),
+      PrimitiveTime.parsePrimitive("time2"),
+    ];
+    sink.options = PrimitiveBoolean.parsePrimitive(false);
+  });
+
+  it("should create a new object, identical to the original", () => {
+    const newSink = sink.clone();
+    expect(newSink).not.toBe(sink);
+    expect(newSink).toEqual(sink);
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=6219961)